internal/uvm/virtual_device.go

```diff
@@ -15,54 +15,106 @@ import (
 // belongs to.
 type VPCIDevice struct {
 	vm *UtilityVM
-	ID string
+	// VMBusGUID is the GUID associated with the instance of a new vmbus child
+	// device, vpci, from which the target assigned device will be attached to
+	// in the UVM. This is used to locate the assigned devices in the UVM's view.
```

Comment:
I am having a lot of trouble parsing this comment. Is `VMBusGUID` just the ID of this device when exposed via VMBus?

Reply:
Yeah, the setup is a bit confusing. Basically, when we assign a device into the UVM, a new bus is created as a child device of vmbus. This bus is the vpci bus. The actual target device (gpu here) is attached as a child to the vpci bus. When you assign the device, you can pass through a guid, referred to as the vmbus guid, that is used to create the vmbus child bus's (vpci) instance ID in wcow and the device path in lcow. In lcow, the device will then show up under /sys/bus/vmbus/devices/<VMBusGUID>/ in the UVM. Does that help at all? I can explain more.

Reply:
From what I understand, I think a clearer phrasing would be "VMBusGUID is the instance ID for this device when it is exposed via VMBus". The fact that there is a VPCI bus created over VMBus, and it's exposed on that, seems like an implementation detail.

Reply:
Fair enough. Done.